### ISSUE TITLE
Infinite loop when removing a card with multiple BeIDCards opened instances

### ISSUE THREAD

**Issue by florian-courault** [Fedict/commons-eid#12]
Hello,

**Description:**
On Linux, using the java-8-openjdk, I get an infinite loop.
When it happens, we cannot use the card anymore, even if we reconnect the terminal and the card.

**Steps to reproduce:**
- Instantiate the BeIDCards class more than one time.
- Remove the card from the terminal.

**Code to reproduce:** 
```java
public static void main(String[] args) {
    Logger logger = new Logger() {
      @Override
      public void error(String s) {
        System.out.println(s);
      }
      @Override
      public void debug(String s) {
        System.out.println(s);
      }
    };
    Scanner sc = new Scanner(System.in);
    while (true) {
      switch (sc.nextInt()) {
        case 0:
          new BeIDCards(logger);
          break;
      }
    }
  }
```
**Cause :**
It happens if we perform multiple JCA actions like sign multiple times; or if we use JCA and non-JCA methods (example : sign and then fetch identity file).
It happens because we instantiate the BeIDCards class in the BeIDKeyStore class, in getBeIDCard, but this instance is never closed.

**Log trace :**
```
10:43:14.595 [CardAndTerminalManager] closing eID card
10:43:14.598 [CardAndTerminalManager] eID Card Removal Reported
10:43:22.709 [CardAndTerminalManager] Cannot wait for card terminal events [2] (No Card Readers Connected?): waitForChange() failed
10:43:22.709 [CardAndTerminalManager] no card readers connected?
10:44:03.322 [CardAndTerminalManager] cause: SCARD_E_NO_SERVICE
10:44:03.322 [CardAndTerminalManager] cause type: sun.security.smartcardio.PCSCException
10:44:03.322 [CardAndTerminalManager] cleared
10:44:09.311 [CardAndTerminalManager] subsystem not initialized
10:44:17.259 [CardAndTerminalManager] Cannot enumerate card terminals [1] (No Card Readers Connected?): list() failed
10:44:17.260 [CardAndTerminalManager] no card readers connected?
10:44:17.261 [CardAndTerminalManager] cause: SCARD_E_NO_SERVICE
10:44:17.261 [CardAndTerminalManager] cause type: sun.security.smartcardio.PCSCException
10:44:23.784 [CardAndTerminalManager] log: cleared
10:44:24.918 [CardAndTerminalManager] subsystem not initialized
10:44:25.352 [CardAndTerminalManager] Cannot enumerate card terminals [1] (No Card Readers Connected?): list() failed
10:44:25.353 [CardAndTerminalManager] no card readers connected?
```
The "list() failed" exception seems to be related to this old ticket: [https://bugs.openjdk.java.net/browse/JDK-8026326](https://bugs.openjdk.java.net/browse/JDK-8026326)
I don't know if there is something we can do about it ...

Let me now what do you think about it or if you need more information !
Regards, 
Florian

PS : I saw your release on maven central, thanks about that !

**Comment by florian-courault** [comment]
We can avoid this problem if we pass a BeIDCard to the keystore with the setBeIDCard method of the BeIDKeyStoreParameter class.

**Comment by florian-courault** [comment]
I discussed with my colleague and he already had this problem with smartcardio, he gave me a solution that I implemented in this PR : #13 . It uses a proxy for the CardTerminals class, which relaunches the service that manages smartcards, using introspection.

**Comment by janvdbergh** [comment]
@florian-courault I added a few cosmetic changes (see https://github.com/Fedict/commons-eid/pull/14). If this is ok for you I will merge and release it.

**Comment by florian-courault** [comment]
This is ok for me.
Again, thank you !

**Comment by janvdbergh** [comment]
Released a new version of eid-commons (1.0.5) which should be available on Maven Central shortly.
